app/screens/settings/Settings.tsx

```diff
@@ -524,12 +526,19 @@ class Settings extends Component<Props, State> {
   toggleSignMessageModal = ({ index }: { index: number }) => {
     this.setState({ signMessageModalAccountIndex: index });
   };
+
+  closeWallet = () => {
+    const { closeWallet, history } = this.props;
+    // @ts-ignore
+    closeWallet();
+    history.push('/auth/unlock');
+  };
 }
 
 const mapStateToProps = (state: RootState) => ({
   displayName: state.wallet.meta.displayName,
   accounts: state.wallet.accounts,
-  walletFiles: state.wallet.walletFiles,
+  walletFiles: state.wallet.walletFiles?.map(({ path }) => path) || null,
```

Comment:
```suggestion
  walletFiles: state.wallet.walletFiles?.map(({ path }) => path) || [],
```
keep same expectation for this field. up to u.

Reply:
I checked in the PR , this field creating a mess around, so we can simplify it and not to use this construction `!walletFiles || walletFiles.length === 0`